new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/LoggingUtil.java

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.weblogic.kubernetes.utils;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import io.kubernetes.client.openapi.ApiException;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+
+import static io.kubernetes.client.util.Yaml.dump;
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+
+/**
+ * A utility class to collect logs for artifacts in Kubernetes cluster.
+ */
+public class LoggingUtil {
+
+  /**
+   * Directory to store logs.
+   */
+  private static final String LOGS_DIR = System.getProperty("RESULT_ROOT",
+        System.getProperty("java.io.tmpdir"));
+
+  /**
+   * Collect logs for artifacts in Kubernetes cluster for current running test object. This method can be called
+   * anywhere in the test by passing the test instance object and list namespaces.
+   * <p>
+   * The collected logs are written in the LOGS_DIR/IT_TEST_CLASSNAME/CURRENT_TIMESTAMP directory.
+   *
+   * @param itInstance the integration test instance
+   * @param namespaces list of namespaces used by the test instance
+   */
+  public static void collectLogs(Object itInstance, List namespaces) {
+    logger.info("Collecting logs...");
+    String resultDirExt = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
+    try {
+      Path resultDir = Files.createDirectories(
+          Paths.get(LOGS_DIR, itInstance.getClass().getSimpleName(),
+              resultDirExt));
+      for (var namespace : namespaces) {
+        LoggingUtil.generateLog((String) namespace, resultDir);
+      }
+    } catch (IOException ex) {
+      logger.warning(ex.getMessage());
+    } catch (ApiException ex) {
+      logger.warning(ex.getResponseBody());
+    }
+  }
+
+  /**
+   * Queries the Kubernetes cluster to get the logs for various artifacts and writes it to the resultDir.
+   *
+   * @param namespace in which to query cluster for artifacts
+   * @param resultDir existing directory to write log files
+   * @throws IOException when writing to log files fail
+   * @throws ApiException when Kubernetes cluster query fails
+   */
+  public static void generateLog(String namespace, Path resultDir) throws IOException, ApiException {
+    logger.info("Collecting logs in namespace : {0}", namespace);
+    // get service accounts
+    writeToFile(Kubernetes.listServiceAccounts(namespace), resultDir.toString(), namespace + "_sa.log");
+    // get namespaces
+    writeToFile(Kubernetes.listNamespacesAsObjects(), resultDir.toString(), namespace + "_ns.log");
+    // get pv
+    writeToFile(Kubernetes.listPersistenVolumes(), resultDir.toString(), namespace + "_pv.log");
```

Comment:
We need some way to label PV's.  This will generate a lot of data since it will grab all PV's, including those associated with domains from other tests.

Reply:
Maybe just grab PV's associated with PVC's we have in the namespace?

Reply:
Grabbing the weblogic.domainUID value from PVC and using that to list PVs.